Code/GraphMol/Descriptors/RDF.cpp

```diff
@@ -0,0 +1,350 @@
+//
+// For more information on the Plane of Best Fit please see
+// http://pubs.acs.org/doi/abs/10.1021/ci300293f
+//
+//  If this code has been useful to you, please include the reference
+//  in any work which has made use of it:
+
+//  Plane of Best Fit: A Novel Method to Characterize the Three-Dimensionality
+//  of Molecules, Nicholas C. Firth, Nathan Brown, and Julian Blagg, Journal of
+//  Chemical Information and Modeling 2012 52 (10), 2516-2525
+
+//
+//
+// Created by Nicholas Firth, November 2011
+// Modified by Greg Landrum for inclusion in the RDKit distribution November
+// 2012
+// Further modified by Greg Landrum for inclusion in the RDKit core September
+// 2016
+// Adding RBF descriptors to 3D descriptors by Guillaume Godin
+
+#include <GraphMol/RDKitBase.h>
+#include <GraphMol/MolOps.h>
+#include <GraphMol/PeriodicTable.h>
+
+#include <GraphMol/MolTransforms/MolTransforms.h>
+
+#include "RDF.h"
+#include "MolData3Ddescriptors.h"
+
+#include <Numerics/Matrix.h>
+#include <Numerics/SquareMatrix.h>
+#include <Numerics/SymmMatrix.h>
+#include <boost/foreach.hpp>
+#include <math.h>
+#include <Eigen/Dense>
+
+// data checked using book Todeschini R., Consonni V. - Molecular Descriptors for Chemoinformatics 2009 atomic properties page 21/22
+
+
+
+namespace RDKit {
+namespace Descriptors {
+namespace {
+
+  MolData3Ddescriptors moldata3D;
+
+
+std::vector<double> getG(int n) {
+  std::vector<double> res(n);
+  for (int i = 0; i < n; i++) {
+    res[i] = 1 + i * 0.5;
+  }
+  return res;
+}
+
+/*
+std::vector<double> CalcChargeRDF(const ROMol &mol, const Conformer &conf) {
+  int numAtoms = conf.getNumAtoms();
+  int confId = conf.getId();
+
+  std::vector<double> R = getG(30);
+  std::vector<double> RDFres;
+  //std::vector<std::vector<double> > DM = GetGeometricalDistanceMatrix(points);
+  double *DM = MolOps::get3DDistanceMat(mol,confId);
+
+  std::vector<double> charges = GetCharges(mol);
+
+  for (int i = 0; i < 30; i++) {
+    double res = 0;
+    for (int j = 0; j < numAtoms - 1; j++) {
+      for (int k = j + 1; k < numAtoms; k++) {
+        res += charges[j] * charges[k] * exp(-100 * pow(R[i] - DM[j * numAtoms + k], 2));
+      }
+    }
+
+      RDFres.push_back(round( 1000 * res) / 1000);
+  }
+
+  return RDFres;
+
+*/
+
+
+std::vector<double> CalcUnweightedRDF(const ROMol &mol, const Conformer &conf) {
+  int numAtoms = conf.getNumAtoms();
+  int confId = conf.getId();
+
+  std::vector<double> R = getG(30);
+
+  std::vector<double> RDFres;
+
+  double *DM = MolOps::get3DDistanceMat(mol,confId);
+
+  for (int i = 0; i < 30; i++) {
+    double res = 0;
+    for (int j = 0; j < numAtoms - 1; j++) {
+      for (int k = j + 1; k < numAtoms; k++) {
+        res += exp(-100 * pow(R[i] - DM[j * numAtoms + k], 2));
+          
+      }
+    }
+
+    RDFres.push_back(round( 1000 * res) / 1000);
+  }
+
+  return RDFres;
+}
+
+
+
+std::vector<double> CalcMassRDF(const ROMol &mol, const Conformer &conf) {
```

Comment:
The same comments I made for the weighted MORSE functions apply here too. A lot of duplicated code